datacake-cluster/src/storage.rs

```diff
@@ -752,6 +784,13 @@ pub mod test_suite {
             .mark_as_tombstone(KEYSPACE, doc_3.id, doc_3.last_updated)
             .await
             .expect("Delete documents from store.");
+        // NOTE: i manually added this call as the test seems to expect the entry in the keyspace
+        //       to have been removed
+        storage
+            .remove_tombstones(KEYSPACE, [3_u64].into_iter())
+            .await
+            .unwrap();
+
```

Comment:
This is incorrect, this should work with the `mark_as_tombstone` without the need for this, Marking a document as a tombstone means the data can be removed entirely but the metadata must stay available when fetching with the metadata methods in order to rebuild the state.

It's incorrect to do this as the store will *not* call remove_tomestones unless it has observed state changes from the node issuing the removal. This is a metadata method as part of the CRDT purge system.